src/test/java/io/confluent/examples/streams/HandlingCorruptedInputRecordsIntegrationTest.java

```diff
@@ -104,34 +104,27 @@ public void shouldIgnoreCorruptInputRecords() {
       //
       // Step 2: Produce some corrupt input data to the input topic.
       //
-      IntegrationTestUtils.produceKeyValuesSynchronously(
-        inputTopic,
-        Collections.singletonList(new KeyValue<>(null, "corrupt")),
-        topologyTestDriver,
-        new IntegrationTestUtils.NothingSerde<>(),
-        new StringSerializer()
-      );
+      topologyTestDriver.createInputTopic(inputTopic,
+                                          new IntegrationTestUtils.NothingSerde<>(),
+                                          new StringSerializer())
+        .pipeInput("corrupt");
 
       //
       // Step 3: Produce some (valid) input data to the input topic.
       //
-      IntegrationTestUtils.produceKeyValuesSynchronously(
-        inputTopic,
-        inputValues.stream().map(v -> new KeyValue<>(null, v)).collect(Collectors.toList()),
-        topologyTestDriver,
-        new IntegrationTestUtils.NothingSerde<>(),
-        new LongSerializer()
-      );
+      topologyTestDriver.createInputTopic(inputTopic,
+                                          new IntegrationTestUtils.NothingSerde<>(),
+                                          new LongSerializer())
+        .pipeValueList(inputValues);
```

Comment:
It seems like declaring the same input topic twice may disguise the intent of this test. Do you think declaring it once with a ByteArray value serializer, and then piping in invalid and valid data arrays would be clearer?